analysis/quantifier-comparisons/08-genes-detected-comparison.Rmd

```diff
@@ -303,6 +306,66 @@ These genes that are found in cr-like-em include the following:
 - ENSG00000248167 - TRIM39-RPP21
 - ENSG00000285508 - ALO34330.1
 
+### Correlation of gene expression between Cellranger and Alevin-fry
+
+We can also look at the correlation of gene expression across these tools and use a linear regression model to see how similar the mean gene expression determined by cellranger is to each of the alevin-fry methods. 
+Here, I am using a linear regression model for each sample and grabbing the residuals for each gene. 
+The residuals will help show us if alevin-fry is over or underestimating gene counts in comparison to cellranger.
+
+```{r}
+# create separate columns for mean log expression for each tool
+mean_exp_df_spread <- mean_exp_df %>%
+  tidyr::pivot_wider(id_cols = c(gene_id, sample),
+                     names_from = alevin_resolution,
+                     values_from = mean_log_expression)
+
+# use a linear regression model to compare gene expression across tools 
+# grab the residuals for each gene and add to the original dataframe
+residuals_df <- mean_exp_df_spread %>%
+  dplyr::group_by(sample) %>%
+  dplyr::mutate(
+    cellranger_cr_like = lm(cellranger ~ `cr-like`)$residuals,
+    cellranger_cr_like_em = lm(cellranger ~ `cr-like-em`)$residuals
```

Comment:
I feel like we should do the comparisons in the opposite direction here, since we are in general treating cellranger as the reference.
```suggestion
    cellranger_cr_like = lm(`cr-like` ~ cellranger )$residuals,
    cellranger_cr_like_em = lm(`cr-like-em` ~ cellranger)$residuals
```